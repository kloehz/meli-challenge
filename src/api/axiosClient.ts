import axios, { AxiosInstance } from "axios";

export const axiosClient: AxiosInstance = axios.create({
    baseURL: 'http://localhost:9000/api'
});