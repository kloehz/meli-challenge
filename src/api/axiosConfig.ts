import axios from "axios";

export const axiosConfig = axios.create({
    baseURL: 'http://localhost:9000/api'
});