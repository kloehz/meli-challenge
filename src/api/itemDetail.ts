import { IGetItemDetail } from "../types/getItemDetail";
import { axiosConfig } from "./axiosConfig";

export const apiGetItemDetails = (id: string) => {
  return axiosConfig.get<IGetItemDetail>(`/items/${id}`);
}
