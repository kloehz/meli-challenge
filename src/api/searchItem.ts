import { IGetItems } from "../types/getItems";
import { axiosConfig } from "./axiosConfig";

export const apiGetItems = (query: string) => {
  return axiosConfig.get<IGetItems>(`/items?q=:${query}`);
}
