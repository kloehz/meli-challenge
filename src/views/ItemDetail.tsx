import { useEffect, useState, } from "react";
import { useLocation } from "react-router-dom";
import { apiGetItemDetails } from "../api/itemDetail";
import { IGetItemDetail, Item } from "../types/getItemDetail";
import { Spinner } from "../components/loader/Spinner";
import useCurrency from "../hooks/useCurrency";

export const ItemDetail = () => {

    const location = useLocation<IGetItemDetail>();
    const [isLoading, setIsLoading] = useState(true);
    const [item, setItem] = useState<Item>();
    const [price, setPrice] = useState<string>('');

    useEffect(() => {
        apiGetItemDetails(location.pathname.slice(7, location.pathname.length)).then((res) => {
            setItem(res.data.item);
            setPrice(() => useCurrency(res.data.item.price));
            setIsLoading(false)
        }).catch((error) => {
            setIsLoading(false);
        });
    }, []);

    return (
            (
                isLoading
                ? <Spinner />
                : <div className="item-detail-page">
                    <div className="item-detail-container">
                            <div className="item-image-price-container">
                                <img src={item.picture}/>
                                <div className="item-detail-to-buy-container">
                                    <span className="item-detail-condition-selled" >
                                        {item.condition} - {item.sold_quantity} vendido(s)
                                    </span>
                                    <h1 className="item-detail-title">
                                        {item.title}
                                    </h1>
                                    <span className="item-detail-price">
                                        {price}
                                    </span>
                                <button className="item-detail-buy-button">Comprar</button>
                                </div>
                            </div>
                            <div className="item-detail-description-container">
                                <div>
                                    <span>
                                        Descripcion del producto
                                    </span>
                                </div>
                                <div>
                                    <span>
                                        {item.description}
                                    </span>
                                </div>
                            </div>
                    </div>
                </div>
            )
    )
}
