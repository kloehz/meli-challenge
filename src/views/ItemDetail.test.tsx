import axios from "axios";
import { shallow } from "enzyme"
import { apiGetItemDetails } from "../api/itemDetail";
import { ItemDetail } from "./ItemDetail";

jest.mock('../api/itemDetail');

jest.mock('react-router-dom', () => ({
    // @ts-ignore
    ...jest.requireActual("react-router-dom"),
    useLocation: jest.fn().mockReturnValue({
        pathname: '/items/MLA912120163'
    })
}))

describe('ItemDetail tests', () => {
    const wrapper = shallow(<ItemDetail />);
    test('should be render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('api apiGetItemDetails should be called', () => {
        // expect(apiGetItemDetails).toHaveBeenCalled();
    })
    
    
})
