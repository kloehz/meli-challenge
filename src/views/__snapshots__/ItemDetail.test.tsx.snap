// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetail tests should be render correctly 1`] = `ShallowWrapper {}`;
